packages/trpc/server/routers/loggedInViewer/connectedCalendars.handler.ts

```diff
@@ -48,6 +48,15 @@ export const connectedCalendarsHandler = async ({ ctx }: ConnectedCalendarsOptio
       So create a default destination calendar with the first primary connected calendar
       */
     const { integration = "", externalId = "", credentialId } = connectedCalendars[0].primary ?? {};
+    // Select the first calendar matching the primary by default since that will also be the destination calendar
+    if (externalId) {
+      const calendarIndex = connectedCalendars[0].calendars.findIndex(
```

Comment:
`connectedCalendars[0].calendars` can be undefined. This causes a type error (also in all other places were `connectedCalendars[0].calendars` is used)

Reply:
If `externalId ` is present, then it means there should be atleast one calendar as per my understanding. Ref: https://github.com/calcom/cal.com/blob/main/packages/core/CalendarManager.ts#L72. I could be wrong though.

Reply:
We still getting type errors though: https://github.com/calcom/cal.com/actions/runs/5043948903/jobs/9046435316?pr=9028

```suggestion
      const calendarIndex = connectedCalendars[0].calendars?.findIndex(
```